internal/store/postgres/session_repository.go

```diff
@@ -1,34 +1,175 @@
+		if nrCtx != nil {
+			nr := newrelic.DatastoreSegment{
+				Product:    newrelic.DatastorePostgres,
+				Collection: TABLE_SESSIONS,
+				Operation:  "Create",
+				StartTime:  nrCtx.StartSegmentNow(),
+			}
+			defer nr.End()
+		}
+
+		return s.dbc.QueryRowxContext(ctx, query, params...).StructScan(&sessionModel)
+	}); err != nil {
+		err = checkPostgresError(err)
+		return fmt.Errorf("%w: %s", dbErr, err)
+	}
+
 	return nil
 }
 
-func (s *SessionRepository) Get(id uuid.UUID) (*authenticate.Session, error) {
-	if val, ok := s.data[id.String()]; ok {
-		return val, nil
+func (s *SessionRepository) Get(ctx context.Context, id uuid.UUID) (*authenticate.Session, error) {
+	var session Session
+	query, params, err := dialect.From(TABLE_SESSIONS).Where(
+		goqu.Ex{
+			"id": id,
+		}).ToSQL()
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	if err := s.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+		nrCtx := newrelic.FromContext(ctx)
+		if nrCtx != nil {
+			nr := newrelic.DatastoreSegment{
+				Product:    newrelic.DatastorePostgres,
+				Collection: TABLE_SESSIONS,
+				Operation:  "Get",
+				StartTime:  nrCtx.StartSegmentNow(),
+			}
+			defer nr.End()
+		}
+
+		return s.dbc.QueryRowxContext(ctx, query, params...).StructScan(&session)
+	}); err != nil {
+		err = checkPostgresError(err)
+		switch {
+		case errors.Is(err, sql.ErrNoRows):
+			return nil, fmt.Errorf("%w: %s", dbErr, authenticate.ErrNoSession)
+		default:
+			return nil, fmt.Errorf("%w: %s", dbErr, err)
+		}
 	}
-	return nil, authenticate.ErrNoSession
+
+	return session.transformToSession(), nil
 }
 
-func (s *SessionRepository) Delete(id uuid.UUID) error {
-	delete(s.data, id.String())
-	return nil
+func (s *SessionRepository) Delete(ctx context.Context, id uuid.UUID) error {
+	query, params, err := dialect.Delete(TABLE_SESSIONS).
+		Where(
+			goqu.Ex{
+				"id": id,
+			},
+		).ToSQL()
+	if err != nil {
+		return fmt.Errorf("%w: %s", queryErr, err)
+	}
+
+	return s.dbc.WithTimeout(ctx, func(ctx context.Context) error {
+		nrCtx := newrelic.FromContext(ctx)
+		if nrCtx != nil {
+			nr := newrelic.DatastoreSegment{
+				Product:    newrelic.DatastorePostgres,
+				Collection: TABLE_SESSIONS,
+				Operation:  "Delete",
+				StartTime:  nrCtx.StartSegmentNow(),
+			}
+			defer nr.End()
+		}
+
+		result, err := s.dbc.ExecContext(ctx, query, params...)
+		if err != nil {
+			err = checkPostgresError(err)
+			switch {
+			case errors.Is(err, sql.ErrNoRows):
+				return fmt.Errorf("%w: %s", dbErr, authenticate.ErrNoSession)
+			default:
+				return fmt.Errorf("%w: %s", dbErr, err)
+			}
+		}
+
+		if count, _ := result.RowsAffected(); count > 0 {
+			return nil
+		}
+
+		return authenticate.ErrDeletingSession
+	})
+}
+
+func (s *SessionRepository) DeleteExpiredSessions(ctx context.Context, logger log.Logger) error {
+	query, params, err := dialect.Delete(TABLE_SESSIONS).
+		Where(
+			goqu.Ex{
+				"expires_at": goqu.Op{"lt": goqu.L("NOW()")},
```

Comment:
I think its better if we pass our expiry time manually if in case db timezone is not matching the application's timezone.